packages/report-server/src/reportBuilder/functions/math.ts

```diff
@@ -7,12 +7,22 @@ import { sum as mathjsSum, divide as mathjsDivide, Matrix, filter } from 'mathjs
 
 const isDefined = <T>(value: T): value is Exclude<T, undefined> => value !== undefined;
 
-export const sum = (numbers?: Matrix): number | undefined => {
-  if (numbers === undefined) {
+export const sum = (
+  ...args: (number | undefined | (number | undefined)[])[]
+): number | undefined => {
+  const numbers = args.flat();
```

Comment:
I was thinking about this recently and part of the issue this causes in older reports is that if a report uses `sum([1, 2, 3])` format, that `[1, 2, 3]` is being passed as a `DenseMatrix` _not_ an array. So there's an additional case to deal with here.

Reply:
That's true! And I think this was the case when I first tried to make this refactor and broke a bunch of lesmis vizes 😢 

I added a migration to this work to re-write those reports to use the new syntax! They seemed to work correctly when testing in local 😄 

However, you raise a good point that we may want to support dense matrix regardless? I've gone back and forth on this... At first I tried to support it, but then began to feel that either:
a) A person is directly writing `sum([a, b, c, ...])` in which case, they shouldn't and should write `sum(a, b, c, ...)` instead
b) We are calling `sum` on a result of a function that returns a dense matrix, eg. `sum(dotProduct([a,b,c], [d,e,f]))`. This case is a bit trickier to handle, but I feel like we'd prefer to ensure that the inner function returns an array rather than a dense matrix. The whole dense matrix thing is a carry over from `mathjs` which we kinda wanna avoid using if possible.

How does this sound to you? I can understand if we wanna support the dense matrix however it does make the code a bit uglier 😢

Reply:
I think that's a fair assessment. Can we add an error message at least? I recall part of the issue when we first tried this was that the sum running against a dense matrix returned undefined sometimes rather than just erroring out.

If we aren't supporting it probably better to have it error out early and descriptively rather than attempting to process it and hoping it will error out at some point right?

Reply:
Mmm yeah good point... I'll add some kind of validator 👍